docs/content/latest/api/ysql/datatypes/type_array/functions-operators/any-all.md

```diff
@@ -0,0 +1,306 @@
+ true
+```
+Notice that here, _"(17, 42, 53)"_ is _not_ a constructed record value because of the semantic effect of preceding it by `IN`. In contrast, in the following example, _"(17, 42, 53)"_ _is_ a constructed record:
+```plpgsql
+with v as (select (17, 42, 53) as r)
+select pg_typeof(r)::text from v;
+```
+This outcome is due to the semantic effect of using _"(17, 42, 53)"_ directly as a `SELECT` list item. This, therefore, causes a syntax error:
+
+```
+with v as (select (1, 2, 3, 4) as r)
+select 1 in r from v;
+```
+
+## The semantics of the =ALL operator
+
+The following rules, governing the result of `=ALL`, can be re-written from the rules that govern the result of `=ANY` by applying your understanding of the difference, in ordinary prose, between the words _any_ and _all_:
+
+- If either the LHS _element_ expression or the RHS array expression evaluates to `NULL`, then the `=ALL` result is `NULL`.
+- If the LHS _element_ is `NOT NULL` and the RHS array has zero _elements_, then the `=ALL` result is `TRUE`. You might think that this is a counter-intuitive rule definition. But the [PostgreSQL documentation for Version 11.2](https://www.postgresql.org/docs/11/functions-comparisons.html#id-1.5.8.28.17) states this clearly. And the first example in the [Semantics demonstration](./#semantics-demonstration) section is consistent with this definition. Moreover, and as is required to be the case, the behavior of the example is identical using YugabyteDB and PostgreSQL Version 11.2.
+- If the succession of comparisons of the LHS _element_ to the RHS array's _elements_ yields no `NULL` results and yields at least one `FALSE` result, then the `=ALL` result is `FALSE`.
+- If the succession of comparisons of the LHS _element_ to the RHS array's _elements_ yields at least one `NULL` result, then the `=ALL` result is `NULL`.
+- In summary, _only_ if every one of the RHS array's _elements_ is `NOT NULL`, all of these _elements_ are equal to each other, and the LHS _element_ is also equal to that value, does the `=ALL` comparison yield `TRUE`.
+
+Here is an example. It shows that you can test whether an array's _elements_ are all the same as each other without needing to know their common value or anything about the array's dimensionality.
+
+```plpgsql
+do $body$
+declare
+  arr constant int[] not null := '
+    [2:3][4:6][7:10]={
+      {
+        {42,42,42,42},{42,42,42,42},{42,42,42,42}
+      },
+      {
+        {42,42,42,42},{42,42,42,42},{42,42,42,42}
+      }
+    }'::int[];
+
+   val constant int not null := (
+    select unnest(arr) limit 1);
+
+  b99 boolean not null := val =ALL (arr);
+begin
+  assert b99, 'assert failed';
+end;
+$body$;
+```
+See the section [Multidimensional array of `int` values](../../literals/array-of-primitive-values/#multidimensional-array-of-int-values) for the explanation of the general syntax for the literal for a multidimensional array that specifies the lower and upper index bounds along each dimension.
```

Comment:
```suggestion
For the explanation of the general syntax for the literal for a multidimensional array that specifies the lower and upper index bounds along each dimension, see [Multidimensional array of `int` values](../../literals/array-of-primitive-values/#multidimensional-array-of-int-values).
```

Reply:
Tell them what they're going to see before you give them a link to it. I think you know this but had a lapse. ;-)

Reply:
Done.